packages/controls/src/controls/combobox/combobox.test.ts

```diff
@@ -0,0 +1,74 @@
+import { testDefinition } from '../../testing/test-definition'
+
+import { Combobox, ComboboxDefinition } from './combobox'
+
+describe('Combobox', () => {
+  describe('constructor', () => {
+    test("definition's config type is derived from constructor's arguments", () => {
+      // Assert
+      Combobox({
+        label: 'Nato',
+        getOptions: () => [
+          { id: 'a', value: 'a', label: 'alpha' },
+          { id: 'b', value: 'b', label: 'beta' },
+        ],
+      }).config satisfies {
+        label?: string
+        getOptions: () => { id: string; value: 'a' | 'b'; label: string }[]
+      }
+
+      Combobox({
+        label: 'Nato',
+        getOptions: async () => [
+          { id: 'a', value: 'a', label: 'alpha' },
+          { id: 'b', value: 'b', label: 'beta' },
+        ],
+      }).config satisfies {
+        label?: string
+        getOptions: () => Promise<
+          { id: string; value: 'a' | 'b'; label: string }[]
+        >
+      }
+    })
+
+    test('disallows extraneous properties', () => {
```

Comment:
Not every control has these tests, should I go ahead and backfill them to the other controls?

Reply:
All of them should:

<img width="413" alt="Screenshot 2024-08-21 at 12 41 58 PM" src="https://github.com/user-attachments/assets/934bdef9-a9ea-43dd-a880-e6940a83dfe8">